File: TransCoder_tokenized_test_set_functions/java/PRINT_WAYS_BREAK_STRING_BRACKET_FORM.java
static void findCombinations ( String str, int index, String out ) {
  if ( index == str . length ( ) ) System . out . println ( out );
  for ( int i = index;
  i < str . length ( );
  i ++ ) findCombinations ( str, i + 1, out + "(" + str . substring ( index, i + 1 ) + ")" );
}

File: TransCoder_tokenized_test_set_functions/python/PRINT_WAYS_BREAK_STRING_BRACKET_FORM.py
def findCombinations ( string, index, out ) :
    if index == len ( string ) :
        print ( out )
    for i in range ( index, len ( string ), 1 ) :
        findCombinations ( string, i + 1, out + "(" + string [ index : i + 1 ] + ")" )
    
